cachi2/core/package_managers/bundler/main.py

```diff
@@ -24,5 +34,61 @@ def fetch_bundler_source(request: Request) -> RequestOutput:
 
 def _resolve_bundler_package(package_dir: RootedPath, output_dir: RootedPath) -> list[Component]:
     """Process a request for a single bundler package."""
-    parse_lockfile(package_dir)
+    dependencies = parse_lockfile(package_dir)
+    _get_main_package_info(package_dir, dependencies)
     return []
+
+
+def _get_main_package_info(
+    package_dir: RootedPath,
+    dependencies: ParseResult,
+) -> tuple[str, Optional[str]]:
+    """Get main package info - name and version."""
+    path_dependencies = [dep for dep in dependencies if isinstance(dep, PathDependency)]
+    name_and_version = _extract_name_and_version_from_gemspec_file(path_dependencies)
+    if name_and_version is not None:
+        return name_and_version
+
+    log.info("Failed to infer name and version from gemspec file")
+
+    # fallback to origin remote
+    try:
+        name = _get_repo_name_from_origin_remote(package_dir)
```

Comment:
Could you please add a  very short comment here or in the extractor function explaining why or when it could raise `UnsupportedFeature`?

Reply:
Awesome, thank you!